pkg/config/controllerconfig.go

```diff
@@ -45,6 +49,9 @@ type ControllerConfig struct {
 	// GCPTokenSource mints OAuth2 tokens to be passed with GCP API calls,
 	// allowing use of a non-default OAuth2 identity
 	GCPTokenSource oauth2.TokenSource
+
+	// EnableMetricsTransport enables automatic wrapping of HTTP clients with metrics transport
+	EnableMetricsTransport bool
```

Comment:
I really am not a fan of bools. We should consider if there is a better enum for this.
MetricsTransport with values "Disabled", "HTTPEnabled" would be better. Then if we add some other transport method we can just add new enumeration values.